app.py

```diff
@@ -227,12 +227,19 @@ def users_entries():
         return redirect(flask.url_for("home"))
     for entry in prev_entries:
         tones.append(get_emotion(entry))
+    """We'll use possible emotions and sort key, to filter the emotions if requested by the user"""
+    possible_emotions = ["All", "Bored", "Fearful", "Excited", "Happy", "Sad"]
+    sort_key = "All"
+    if request.method == "POST":
```

Comment:
If it is a post request, meaning that we have selected one of the filters, we will re organize the entries and tones, so that only the entries/tones that is important to us are shown to the user.